src/App.js

```diff
@@ -16,31 +80,45 @@ const genres = [
 ]
 
 function App() {
-    const [selected, setSelected] = useState(genres[0].name);
+    const [selectedGenre, setSelectedGenre] = useState(genres[0].name);
+	const [selectedMovie, setSelectedMovie] = useState();
+	const [sortedBy, setSortedBy] = useState();
 
-	const onSelect = (event) => {
-		event.preventDefault()
-		setSelected(event.target.dataset.name)
+	const sortMovies = () => {
+		if (sortedBy === 'title') {
+			return movies.sort((prev, next) => prev.name.localeCompare(next.name))
+		}
+	
+		return movies.sort((prev, next) => prev.releaseYear - next.releaseYear)
 	}
 
-	const onSearch = (event, setResult) => {
-		event.preventDefault()
-
-		const inputEl = event.target.querySelector('input')
+	const moviesList = sortedBy ? sortMovies() : movies
 
-		setResult(inputEl.value)
-		inputEl.value = ''
+	const onSelect = (event) => {
+		event.preventDefault()
+		setSelectedGenre(event.target.dataset.name)
 	}
 
   	return (
 		<div className="App">
-			<Counter initialValue={0} />
-			<SearchForm initialValue="Initial search query" handleSubmit={onSearch} />
-			<GenreSelect
-				genres={genres}
-				selected={selected}
-				onClick={onSelect}
-			/>
+			<Header movieDetails={selectedMovie}/>
+			<div className='main'>
+				<div className='d-flex mb-2 nav'>
+					<GenreSelect
+						genres={genres}
+						selected={selectedGenre}
+						onClick={onSelect}
+					/>
+					<SortControl setSortedBy={setSortedBy} />
+				</div>
+				<p className="text-left mb-2"><b>{movies?.length}</b> movies found</p>
+				<div className="d-flex">
+					{moviesList.map((movie) =>
+						<MovieTile key={movie.id} details={movie} selectMovie={setSelectedMovie}/>
```

Comment:
can't see onClick handler mentioned in the task description "Additionally, the component should receive a callback property to capture click event."